client/session.go

```diff
@@ -660,13 +659,14 @@ func (s *session) setTopologyWithLock(topoMap topology.Map, queues []hostQueue,
 	s.fetchBatchOpArrayArrayPool.Init()
 
 	if s.iteratorArrayPool == nil {
-		s.iteratorArrayPool = encoding.NewIteratorArrayPool([]pool.Bucket{
```

Comment:
Hm, any reason to remove this pool?  We used to be able to pool these slice allocations, however this change removes the ability for those to be pooled?

Reply:
s.iteratorArrayPool is changed to be a MultiReaderIteratorPool rather than a generic IteratorPool. I figured it would be better to keep the size the same as previously, and keep it as a separate pool from the other MultiReaderIteratorPool to ensure we have the same amount of pooled iterators